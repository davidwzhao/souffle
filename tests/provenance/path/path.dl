.pragma "provenance" "1"

.decl edge(x:symbol, y:symbol)
edge("a", "b").
edge("b", "c").
edge("c", "d").

.decl path(x:symbol, y:symbol)
path(x, y) :- edge(x, y).
path(x, z) :- edge(x, y), path(y, z).
.output path()
