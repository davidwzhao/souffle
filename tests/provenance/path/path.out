Explain is invoked.
Enter command >                       edge(c, d)               
                      -----------(R0)          
           edge(b, c)   path(c, d)             
           ---------------------------(R1)     
edge(a, b)           path(b, d)                
-------------------------------------------(R1)
                  path(a, d)                   
Enter command > 